Source/Parallelization/GuardCellManager.cpp

```diff
@@ -89,17 +92,29 @@ guardCellManager::Init (
     ng_alloc_J = IntVect(ngJx,ngJz);
 #endif
 
-    // Rho is needed both at the beginning and at the end of the PIC iteration.
-    // Hence we add one extra guard cell for the allocation of the MultiFab that
-    // contains rho, in order to account for the change in the particle positions
-    // within a time step
+    // TODO Adding one cell for rho should not be necessary, given that the number of guard cells
+    // now takes into account the time step (see code block below). However, this does seem to be
+    // necessary in order to avoid some remaining instances of out-of-bound array access in
+    // simulations with large time steps (revealed by building WarpX with BOUND_CHECK = TRUE).
     ng_alloc_Rho = ng_alloc_J+1;
```

Comment:
As I explain in this comment that I added in the code itself (as a reminder), I expected that it would be possible to modify the assignment `ng_alloc_Rho = ng_alloc_J+1;` with `ng_alloc_Rho = ng_alloc_J;`, hence without adding the extra cell for rho.

The extra cell seemed to be there in order to take into account the fact that rho is needed both at the beginning and at the end of the PIC iteration. Now, this should be taken into account automatically with the code added below, that uses the time step (full step `dt` for rho and half step `dt/2` for J) to add the necessary number of extra guard cells based on the time step.

However, when I tried to run the test mentioned in the PR description, using the averaged Galilean PSATD algorithm and a large time step (`warpx.cfl = 3.0`), after compiling WarpX with `BOUND_CHECK = TRUE`, I noticed that the original extra cell for rho seems to be necessary in order to avoid some remaining instances of out-of-bound array access occurring in the charge deposition.

Therefore, even though I do not fully understand why the original extra cell is needed, I left it in place, preventing those instances of out-of-bound array access from happening. Note that such instances would occur silently if WarpX were compiled with `BOUND_CHECK = FALSE`: in this case, they would not seem to affect the results in any significant way.